go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -91,3 +91,8 @@ type Plugin interface {
 	// Properties desired by the plugin
 	GetProperties() PluginProperties
```

Comment:
I was thinking and wdyt of having`PluginCleanupPolicy` defined in `PluginProperties`?

Reply:
that sounds reasonable to me! @EngHabu wdyt?

Reply:
I would rather keep properties to a safe-to-serialize struct (so that it can be read from config...) we can add another optional (`GetOverrides() PluginOverrides`) and that would be where we add all of these functions people can choose to implement or leave empty...